src/Propulsion.Cosmos/PropulsionCosmosPrometheus.fs

```diff
@@ -4,111 +4,145 @@ namespace Propulsion.CosmosStore.Prometheus
 namespace Propulsion.Cosmos.Prometheus
 #endif
 
+/// This file implements a Serilog Sink `LogSink` that publishes metric values to Prometheus.
+/// It takes in an additional set of custom tags to annotate the metric we're publishing.
```

Comment:
I think this comment might better go on the ctors to help decide which one to use